site/gatsby-site/src/components/forms/SubmissionWizard/StepOne.js

```diff
@@ -0,0 +1,337 @@
+import { Badge, Button, Spinner } from 'flowbite-react';
+import { Formik, Form, useFormikContext } from 'formik';
+import React, { useEffect, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import TextInputGroup from '../TextInputGroup';
+import * as yup from 'yup';
+import Label from '../Label';
+import FlowbiteSearchInput from '../FlowbiteSearchInput';
+import RelatedIncidents from 'components/RelatedIncidents';
+import { dateRegExp } from 'utils/date';
+import StepContainer from './StepContainer';
+import TagsInputGroup from '../TagsInputGroup';
+import { Editor } from '@bytemd/react';
+import SemanticallyRelatedIncidents from 'components/SemanticallyRelatedIncidents';
+import IncidentIdField from 'components/incidents/IncidentIdField';
+
+const StepOne = (props) => {
+  const stepOneValidationSchema = yup.object().shape({
+    title: yup
+      .string()
+      .min(6, '*Title must have at least 6 characters')
+      .max(500, "*Titles can't be longer than 500 characters")
+      .required('*Title is required')
+      .nullable(),
+    authors: yup
+      .string()
+      .min(3, '*Authors must have at least 3 characters')
+      .max(200, "*Authors can't be longer than 200 characters")
+      .required('*Author is required. Anonymous or the publication can be entered.')
+      .nullable(),
+    date_published: yup
+      .string()
+      .matches(dateRegExp, '*Date is not valid, must be `YYYY-MM-DD`')
+      .required('*Date published is required')
+      .nullable(),
+    date_downloaded: yup
+      .string()
+      .matches(dateRegExp, '*Date is not valid, must be `YYYY-MM-DD`')
+      .required('*Date downloaded required')
+      .nullable(),
+    url: yup
+      .string()
+      .url('*Must enter URL in http://www.example.com format')
+      .required('*URL required')
+      .nullable(),
+    text: yup
+      .string()
+      .min(80, `*Text must have at least 80 characters`)
+      .max(50000, `*Text can’t be longer than 50000 characters`)
+      .required('*Text is required'),
+    incident_id: yup.number().positive().integer('*Must be an incident number or empty'),
+    incident_date: yup.date().when('incident_id', {
+      is: (incident_id) => incident_id == '' || incident_id === undefined,
+      then: yup.date().required('*Incident Date required').nullable(),
+    }),
+  });
+
+  const handleSubmit = (values, last = false) => {
+    props.next(values, last);
+  };
+
+  return (
+    <StepContainer name={props.name}>
+      <Formik
+        initialValues={props.data}
+        onSubmit={() => {}}
```

Comment:
This doesn't seem right

Reply:
Formik's onSubmit is required, but I can't use it in this case because I can't pass custom data. And in order to know if I'm submitting for a Next Step or for the Final Submit I need to pass a custom param.
Since Formik requires an onSubmit function I need to pass an empty function.